### DISCUSSION TITLE
3d array to store a series of EBSD measurements

### DISCUSSION THREAD

**Discussion by nargesmbg** [mtex-toolbox/mtex#1942] [Ask Anything]
hi Everone, 

I have 27 different samples and try to run automatic analysis on them with for loop. my question is which type of 3D array we need to define to  import ebsd data and calculate grains? could you pleae give me a code which run with for example for loop read 5 ebsd data, and calculate grains? 

Best 
Narges

**Comment by ralfHielscher** [comment]
You can do the following
```matlab
% get all .ang files in the current directory 
files = dir('*.ctf');
for k = 1:length(files)
 ebsd{k} = EBSD.load(files(k).name);
  [grains{k}, ebsd{k}.grainId] = calcGrains(ebsd{k}('indexed');
end
```
Ralf.
